### ISSUE TITLE
🐛  Navigate feature doesn't work with `less -rX` (necessary for hyperlinks feature)

### ISSUE THREAD

**Issue by HarrisonMc555** [dandavison/delta#615]
In order to use the "hyperlinks" feature, the help section suggests setting `DELTA_PAGER` to `less -rX`. This allows hyperlinks to work (instead of being displayed as text).

However, this doesn't play nicely with the "navigate" feature. It appears that the navigate feature gives `less` a regular expression to search for (I've used something similar myself in the past). This regular expression anchors the match to the beginning of the string with `^`. When you use `less -RX`, this works fine. However, when you use `less -rX`, it no longer works. I don't know of a great way to debug this, but I assume that `less` is matching on the escape sequence used to create the colors for each part (`commit`, `Δ`, etc.), and thus the regular expression `^(commit|Δ|...)` no longer matches.

Unfortunately, I don't think I have a great idea of how to fix that 🙁 Maybe it would be feasible to add the color escape sequence to the beginning of the regular expression? And use `?` to make it optional in the regular expression?

Or maybe we just acknowledge that "hyperlinks" and "navigate" can't be used together :shrug:

**Comment by HarrisonMc555** [comment]
Yeah that works great! I didn't read closely enough and was still trying to use `less -rX` at first, but once I switched to `less -RX` with `less` version 581 it worked great! Thanks for the tip :+1:

So I guess the recommendation for hyperlinks now can be something like this? Either 1) use `less -rX`, which breaks the "navigate" feature, or 2) download `less` version >= 581 and use `less -RX` like normal.

**Comment by dandavison** [comment]
Great. Agreed, I've updated the docs in #616.

If you _are_ interested in improving delta's hyperlinks support, there are some open issues containing ideas people have had, and I'd be happy to chat about what direction makes sense in any of them or point in the direction of the code changes that would be involved.

- https://github.com/dandavison/delta/issues/291  (`git blame`)
- https://github.com/dandavison/delta/issues/602 (`--name-only`)
- https://github.com/dandavison/delta/issues/603 (`git log` formats and `--oneline`)

**Comment by dandavison** [comment]
By the way, related to file path links, in master I have added a new option `--relative-paths` which causes delta to always emit file paths relative to the current directory. This can be helpful for file path links.

Maybe worth mentioning also that in the VS Code integrated terminal (which I've recently started using quite a bit) `--relative-paths` is all that is needed to have all file paths emitted by delta open in the editor on ctrl-click / cmd-click (i.e. `--hyperlinks` tricks not needed in that case).
